package_test.go

```diff
@@ -1407,3 +1373,22 @@ func (s *PackageSuite) TestRaceConditionFinalizerTX(c *C) {
 	// Assert that sql.Stmt was not closed early.
 	c.Assert(q.Run(), IsNil)
 }
+
+func (s *PackageSuite) TestStatementTXReuse(c *C) {
+	sqldb, err := setupDB()
+	c.Assert(err, IsNil)
+
+	db := sqlair.NewDB(sqldb)
+
+	// Create a statement and run it on a transaction.
+	selectStmt := sqlair.MustPrepare(`SELECT 'hello'`)
+	tx, err := db.Begin(nil, nil)
+	c.Assert(err, IsNil)
+	q := tx.Query(nil, selectStmt)
+	c.Assert(q.Run(), IsNil)
+	tx.Commit()
+
+	// Run the same existing statement outside the transaction.
+	q = db.Query(nil, selectStmt)
+	c.Assert(q.Run(), IsNil)
```

Comment:
We should close the DB at the end of the test.

Reply:
What do you mean by closing the DB? When the db object gets out of scope the finalizer will be called closing the underlying sql db.

Reply:
Just had another look at it, I meant drop the tables but actually this test doesn't create and tables so it should be ok.